backend/app/core/api_views.py

```diff
@@ -4,22 +4,22 @@
 from core.models.consumer import Consumer
 from django.http.response import HttpResponseBadRequest
 from rest_framework.response import Response
-from rest_framework.permissions import IsAuthenticated
 from django.core import exceptions
-from google.oauth2 import id_token
-from google.auth.transport import requests
-from core.utils import create_consumer
-from app.settings import GOOGLE_CLIENT_ID
+from core.utils import create_consumer,get_consumer_email
 
 HTTP_AUTHORIZATION = "HTTP_AUTHORIZATION"
 
 
 class ConsumerView(APIView):
 
     def post(self, request: Request):
-        token = request.META.get(HTTP_AUTHORIZATION).split(" ")[1]
-        idinfo = id_token.verify_oauth2_token(token, requests.Request(), GOOGLE_CLIENT_ID)
-        email = idinfo['email']
+        http_authorization = request.META.get(HTTP_AUTHORIZATION)
+        
+        try:
+            email = get_consumer_email(http_authorization)
+        except Exception:
+            return Response(status=401)
```

Comment:
Consider catching the specific exception that the util function raises rather than a general exception.